dot/state/block.go

```diff
@@ -354,12 +355,16 @@ func (bs *BlockState) GetBlockBody(hash common.Hash) (*types.Body, error) {
 		return nil, err
 	}
 
-	return types.NewBody(data), nil
+	return types.NewBodyFromBytes(data)
 }
 
 // SetBlockBody will add a block body to the db
 func (bs *BlockState) SetBlockBody(hash common.Hash, body *types.Body) error {
-	return bs.db.Put(blockBodyKey(hash), body.AsOptional().Value())
+	optionalBody, err := body.AsOptional()
+	if err != nil {
+		return err
+	}
+	return bs.db.Put(blockBodyKey(hash), optionalBody.Value())
```

Comment:
I think this can be simplified to just store the body directly, as the value of the optional is the same as `body`:
```suggestion
	return bs.db.Put(blockBodyKey(hash), body))
```